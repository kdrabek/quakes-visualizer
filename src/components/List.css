.list {
    grid-column: 1;
    grid-row: 2 / 4;
    max-height: auto;
    overflow-x: auto;
    background: #f3f6f8;
  }
  