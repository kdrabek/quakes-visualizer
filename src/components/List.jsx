import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import ListItem from './Item';

import styles from './List.css';

const cx = classNames.bind(styles);
const className = cx({
  main: true,
});


class ControlledExpansionPanels extends React.Component {
  constructor(props) {
    super(props);
    this.state = { expanded: null };
    this.handleChange.bind(this);
  }

  handleChange = panel => (event, expanded) => {
    this.setState({ expanded: expanded ? panel : false });
    this.props.updateMap(panel);  //eslint-disable-line
  };

  render() {
    const { earthquakesList } = this.props;  //eslint-disable-line

    return (
      <div className={className}>
        {
          earthquakesList.isEmpty() ?
          'No Events Found' :
          earthquakesList.map((earthquake) => {
            const id = earthquake.get('id');
            return (
              <ListItem
                id={id}
                properties={earthquake.get('properties')}
                handleChange={this.handleChange}
                expanded={this.state.expanded === id}
              />
            );
          })
        }
      </div>
    );
  }
}

ControlledExpansionPanels.propTypes = {
  classes: PropTypes.object.isRequired,  //eslint-disable-line
};

export default ControlledExpansionPanels;
