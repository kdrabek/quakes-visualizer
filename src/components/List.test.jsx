import Immutable from 'immutable';
import { shallow } from 'enzyme';
import List from './List';

const earthquakesList = Immutable.fromJS([
  {
    id: 'some-id',
    properties: { mag: 4, depth: 3.4 },
  },
]);

describe('List component', () => {
  it('should match snapshot when no earthquakes', () => {
    const props = {
      earthquakesList: Immutable.fromJS([]),
    };
    const wrapper = shallow(<List {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot earthquakesList', () => {
    const props = {
      earthquakesList,
      updateMap: () => {},
    };
    const wrapper = shallow(<List {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
