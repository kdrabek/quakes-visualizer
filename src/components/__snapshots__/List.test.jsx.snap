// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should match snapshot earthquakesList 1`] = `
<List
  earthquakesList={
    Immutable.List [
      Immutable.Map {
        "id": "some-id",
        "properties": Immutable.Map {
          "mag": 4,
          "depth": 3.4,
        },
      },
    ]
  }
  updateMap={[Function]}
/>
`;

exports[`List component should match snapshot when no earthquakes 1`] = `
<renderComponent(Empty)
  earthquakesList={Immutable.List []}
/>
`;
